DFReader.py

```diff
@@ -1207,6 +1263,42 @@ def last_timestamp(self):
         m = self.recv_msg()
         return m._timestamp
 
+    def __getstate__(self):
```

Comment:
if would be nice if we could avoid the duplication either by inheriting from a common class, or create a FileHandle class to hold the filehandle that has the pickle hooks on it

Reply:
a FileHandle class would be nicer, as we could use that for tlogs as well, in mavlogfile() class

Reply:
Yes that would work and be better encapsulated as we wouldn't need the additional attributes I added to the DRReader class. We'll need a similar class to provide pickle support for mmap.

Where is the best place to add these classes? I can add them to an existing utils type module or create a new one if preferred.

Reply:
hmm, it's tempting to put it in MAVProxy/modules/lib/, and do this:

 - before we fork:  mlog.filehandle = WrapFileHandle(mlog.filehandle)
 - in the child: mlog.filehandle = mlog.filehandle.unwrap()

that keeps all of the pickling and OS dependent code in MAVProxy

Reply:
I've made the changes here: https://github.com/ArduPilot/MAVProxy/pull/840. I think the approach is in line with what you have in mind. There are a couple of things at least that will need changing:

- the custom pickle code / classes will need a different home, I've kept them in the MAVExplorer file for now
- I think a lock of some sort is required in the code block where child process is spawned. For a couple of lines mestate.mlog has different types for the filehandle and data_map that will break other threads that attempt to read at that point.